magda-gateway/src/loadTenantsTable.ts

```diff
@@ -0,0 +1,13 @@
+import RegistryClient from "../../magda-typescript-common/src/registry/RegistryClient";
+import { Tenant } from "../../magda-typescript-common/src/generated/registry/api";
+
+export default function loadTenantsTable(
+    tenantsTable: Map<String, Tenant>,
+    url: string
+) {
+    let registryClient = new RegistryClient({ baseUrl: url });
+    registryClient.getTenants().then(tenants => {
```

Comment:
Now that I'm seeing this code, it does seem like the tenant details should be their own microservice/DB, rather than living in the registry. If for no other reason than to avoid giving the already-overloaded registry another responsibility.

Reply:
Yeah - I kinda thought we'd talked about putting them in the auth db/api 🤔 .

A new microservice would work too.

Reply:
I'd suggest that we consider this in the future as we are making minimal changes at the moment.